### ISSUE TITLE
Implement a new controller/dispatcher to support NATS JetStream

### ISSUE THREAD

**Issue by k15r** [kyma-project/kyma#13223]
## Acceptance
- [x] #13251
- [x] #13316
- [x] #13321
- [x] #13437
- [x] #13438
- [x] #13486
- [x] #13487
- [x] #13554
- [x] #13544 
- [x] #13546
- [x] #13625
- [x] #13679
- [x] #13739
- [x] #13742
- [x] #13777

**Comment by pxsalehi** [comment]
To install Kyma with the JetStream backend enabled:
```
kyma deploy --value global.features.enableJetStream=true
```

**Comment by pxsalehi** [comment]
To run the Eventing integration tests with the JetStream backend enabled, you'd need a PR with the following content: https://github.com/kyma-project/test-infra/pull/4936/commits/d9aa2601ea3918a8615a06b2ea308490d80cddb2

**Comment by pxsalehi** [comment]
(After #13437) Installing Kyma with the JetStream flag enabled creates the stream. You can use/inspect it with the [NATS CLI](https://github.com/nats-io/natscli).

```
# Port-forward the NATS server
kubectl -n kyma-system port-forward svc/eventing-nats 4222

# List the created stream
nats stream ls

# Subscribe (blocks)
nats sub kyma-eventing.order.created.v1

# Publish
nats pub kyma-eventing.order.created.v1 "order created message"
```

**Comment by raypinto** [comment]
(After #13497) Creating a subscription creates consumers. You can use/inspect it with the [NATS CLI](https://github.com/nats-io/natscli).

```
# Port-forward the NATS server
kubectl -n kyma-system port-forward svc/eventing-nats 4222

# List the created consumers
nats consumer ls

**Comment by mfaizanse** [comment]
New Issue: https://github.com/kyma-project/kyma/issues/13962
